forui/lib/src/widgets/header/root_header.dart

```diff
@@ -0,0 +1,156 @@
+part of 'header.dart';
+
+/// A root header.
+///
+/// A root header contains the page's title and actions.
+/// It is typically used on pages at the root of the navigation stack.
+///
+/// See:
+/// * https://forui.dev/docs/header for working examples.
+/// * [FRootHeaderStyle] for customizing a header's appearance.
+final class _FRootHeader extends FHeader {
+  /// The header's style. Defaults to [FThemeData.headerStyle].
+  final FRootHeaderStyle? style;
+
+  /// The title, aligned to the left.
+  final Widget title;
+
+  /// The actions, aligned to the right. Defaults to an empty list.
+  final List<Widget> actions;
+
+  /// Creates a [FHeader].
+  const _FRootHeader({
+    required this.title,
+    this.style,
+    this.actions = const [],
+    super.key,
+  }) : super._();
+
+  @override
+  Widget build(BuildContext context) {
+    final style = this.style ?? context.theme.headerStyle.rootStyle;
+
+    return SafeArea(
+      bottom: false,
+      child: Padding(
+        padding: style.padding,
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Expanded(
+              child: DefaultTextStyle.merge(
+                overflow: TextOverflow.fade,
+                maxLines: 1,
+                softWrap: false,
+                style: style.titleTextStyle,
+                child: title,
+              ),
+            ),
+            _InheritedActionStyle(
+              style: style.actionStyle,
+              child: Row(children: actions.expand((action) => [action, const SizedBox(width: 10)]).toList()),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty('title', title))
+      ..add(IterableProperty('actions', actions));
+  }
+}
+
+/// [_FRootHeader]'s style.
+final class FRootHeaderStyle with Diagnosticable {
```

Comment:
`FRootHeaderStyle` and `FNestedHeaderStyle` seem to contain the same fields. They only differ in the values initialized by the `inherit(...)` constructor. Perhaps we could move the initialization of fields with different values up to `FHeaderStyles`? That way we can just have different instances of the same type.

Reply:
I disagree, seems like a bad abstraction to me. They inherently represent very different things that have similar fields at this point in time.

I think merging nested and root header to a sealed class to debatable IMO. But merging the style classes completely doesn't make sense to me.